main/src/ca/uwaterloo/flix/language/phase/Namer.scala

```diff
@@ -1357,6 +1373,14 @@ object Namer {
     case WeededAst.Pattern.Cst(Ast.Constant.Null, loc) => throw InternalCompilerException("unexpected null pattern", loc)
     case WeededAst.Pattern.Tag(qname, p, loc) => freeVars(p)
     case WeededAst.Pattern.Tuple(elms, loc) => elms flatMap freeVars
+    case WeededAst.Pattern.Record(pats, pat, _) => recordPatternFreeVars(pats, pat)
+  }
+
+  /**
+    * Returns the free variables in the list of [[Record.RecordFieldPattern]] `pats` and `pat`.
+    */
+  private def recordPatternFreeVars(pats: List[Record.RecordFieldPattern], pat: Option[WeededAst.Pattern]): List[Name.Ident] = {
+    pats.map(_.pat.map(freeVars)).filter(_.isDefined).flatMap(_.get) ++ pat.map(freeVars).getOrElse(Nil)
```

Comment:
Oof. Can this not be made better? _.isDefined? :O

Reply:
What about this
```scala
pats.filter(_.pat.isDefined).flatMap(rfp => freeVars(rfp.pat.get)) ++ pat.map(freeVars).getOrElse(Nil)
```